sonarts-core/src/utils/cognitiveComplexity.ts

```diff
@@ -125,51 +134,55 @@ class ComplexityWalker extends TreeVisitor {
 
   public visitContinueStatement(node: ts.ContinueStatement): void {
     if (node.label) {
-      this.addComplexityWithoutNesting();
+      this.addComplexityWithoutNesting(findChild(node, ts.SyntaxKind.ContinueKeyword));
     }
     super.visitContinueStatement(node);
   }
 
   public visitBreakStatement(node: ts.BreakStatement): void {
     if (node.label) {
-      this.addComplexityWithoutNesting();
+      this.addComplexityWithoutNesting(findChild(node, ts.SyntaxKind.BreakKeyword));
     }
     super.visitBreakStatement(node);
   }
 
   public visitCatchClause(node: ts.CatchClause): void {
-    this.addComplexityWithNesting();
+    this.addComplexityWithNesting(findChild(node, ts.SyntaxKind.CatchKeyword));
     this.visitWithNesting(node.block);
   }
 
   public visitForStatement(node: ts.ForStatement): void {
-    this.visitLoop(node.statement, node.initializer, node.condition, node.incrementor);
+    this.visitLoop(node.statement, ts.SyntaxKind.ForKeyword, node.initializer, node.condition, node.incrementor);
   }
 
   public visitForInStatement(node: ts.ForInStatement): void {
-    this.visitLoop(node.statement, node.initializer, node.expression);
+    this.visitLoop(node.statement, ts.SyntaxKind.ForKeyword, node.initializer, node.expression);
   }
 
   public visitForOfStatement(node: ts.ForOfStatement): void {
-    this.visitLoop(node.statement, node.initializer, node.expression);
+    this.visitLoop(node.statement, ts.SyntaxKind.ForKeyword, node.initializer, node.expression);
   }
 
   public visitDoStatement(node: ts.DoStatement): void {
-    this.visitLoop(node.statement, node.expression);
+    this.visitLoop(node.statement, ts.SyntaxKind.DoKeyword, node.expression);
   }
 
   public visitWhileStatement(node: ts.WhileStatement): void {
-    this.visitLoop(node.statement, node.expression);
+    this.visitLoop(node.statement, ts.SyntaxKind.WhileKeyword, node.expression);
   }
 
-  private visitLoop(nestedNode: ts.Node, ...notNestedExpressions: (ts.Node | undefined)[]) {
-    this.addComplexityWithNesting();
+  private visitLoop(
+    nestedNode: ts.Node,
+    complexityKeyword: ts.SyntaxKind,
+    ...notNestedExpressions: (ts.Node | undefined)[]
+  ) {
+    this.addComplexityWithNesting(findChild(nestedNode.parent!, complexityKeyword));
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Forbidden non null assertion ([typescript:S2966](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS2966))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=issue-244&issues=AWCije4__Q-0dz4JAh0U&open=AWCije4__Q-0dz4JAh0U)

[ISSUE_KEY]: AWCije4__Q-0dz4JAh0U

Reply:
This issue has been resolved by a user in SonarQube

[ISSUE_STATUS]: RESOLVED